action.yaml

```diff
@@ -0,0 +1,60 @@
+---
+name: Run updatecli and push to git
+author: PrefectHQ
+description: This action will run updatecli and push the changes to git and will also open a PR
+inputs:
+  manifest-path:
+    description: "Path to the updatecli manifest file"
+    default: ".github/updatecli/manifest-minor.yaml"
+    required: false
+  run-type:
+    description: "The type of updatecli run to perform (major or minor)"
+    default: "minor"
+    required: false
+runs:
+  using: composite
+  steps:
+    - id: configure_git
+      run: |
+        git config user.name "$GITHUB_ACTOR"
+        git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
+      shell: bash
+
+    - name: Get current date
+      id: date
+      run: echo "date=$(date +'%Y-%m-%d')" >> $GITHUB_OUTPUT
+      shell: bash
+
+    - name: create branch for dependency version updates
+      run: git checkout -b "dependency-version-${{ inputs.run-type }}-${{ steps.date.outputs.date }}"
+      shell: bash
+
+    - name: install updatecli in the runner
+      uses: updatecli/updatecli-action@v2
+
+    - name: run updatecli in apply mode
+      id: updatecli_apply
+      run: |
+        updatecli apply --config .github/updatecli/manifest.yaml --experimental
+        if [[ $(git diff --name-only | wc -l) -eq 0 ]]; then
+          echo "No changes detected, exiting"
+          echo "changes=false" >> $GITHUB_OUTPUT
+          exit 0
+        else
+          echo "changes=true" >> $GITHUB_OUTPUT
+        fi
+        git add .
+        git commit -m "dependency-version-${{ inputs.run-type }}-${{ steps.date.outputs.date }}"
+        git push --set-upstream origin "dependency-version-${{ inputs.run-type }}-${{ steps.date.outputs.date }}"
+      env:
+        GITHUB_TOKEN: ${{ github.token }}
```

Comment:
does this need to be passed in as an action var? i forget if custom actions have access to the workflow's token scope

Reply:
lemme verify you may be right there

Reply:
https://docs.github.com/en/actions/security-guides/automatic-token-authentication#example-1-passing-the-github_token-as-an-input passing like this example.  May be unnecessary but worth it to be confident the right perms are passing